pkg/controller/common/tracing/round_tripper.go

```diff
@@ -0,0 +1,252 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package tracing
+
+import (
+	"bytes"
+	"compress/gzip"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"path"
+	"strings"
+	"sync/atomic"
+	"unsafe"
+
+	"go.elastic.co/apm"
+	"go.elastic.co/apm/module/apmhttp"
+)
+
+// This purely exists as workaround until the issue around closing connections in apm-go-agent is fixed.
```

Comment:
Can you highlight what is the original code and what is the fix?

Reply:
Yes absolutely.